### ISSUE TITLE
cannot play back with latest tvheadend server on master - 403 unauthorized error

### ISSUE THREAD

**Issue by dreamcat4** [zipleen/tvheadend-iphone-client#278]
Hi, using latest tvheadend server version from git master branch:

```sh
HTS Tvheadend 4.3-1994~gc7b713edb-dirty
```

it's a modern iPad, running a recent version of iOS. Using following version of your app... (downloaded from apple app store)

```
tvhclient version 4.0.0 (908) [git:20afbed] build time Sun Jun 28 16:15:06 BST 2020
iOS version 15.2.1
iPad 7th generation
```

so looking at the date when last modified. it seems like the app has not been updated for about 18 months. So maybe there are newer changes on the server side which broke something? This is the only reason i can seem to find an explanation for this happening.

Since have now exhausted all other ideas. Thought I should raise an issue about it.

problem description
===================

trying to open a channel
this result occurs with both the internal vlc player, and with external vlc player

server side log:

```sh
[ERROR] http: 192.168.1.38: HTTP/1.0 GET (1) /stream/channel/065fa7ac302ba277a71b5f9a0168a304 -- 403
```

this occurs with ios player

server side log:

```sh
2022-01-30 10:09:08.008 [   INFO] subscription: 1D2B: "HTTP" subscribing on channel "BBC Parliament", weight: 100, adapter: "SAT>IP DVB-S Tuner #4 (192.168.1.22@UDP)", network: "DVB-S/S2 - 28.2E:Astra", mux: "10788V", provider: "BSkyB", service: "BBC Parliament", profile="pass", hostname="192.168.1.38", username="tv", client="AppleCoreMedia/1.0.0.19C63 (iPad; U; CPU OS 15_2_1 like Mac OS X; en_gb)"
2022-01-30 10:09:08.557 [   INFO] subscription: 1D2B: "HTTP" unsubscribing from "BBC Parliament", hostname="192.168.1.38", username="tv", client="AppleCoreMedia/1.0.0.19C63 (iPad; U; CPU OS 15_2_1 like Mac OS X; en_gb)"
```

the password has no special characters in it. i was wondering what i am doing wrong here. if it was some misconfiguration. However i really cannot find any reason for why.

when vlc opens externally the full url given is as follows:

```sh
vlc://http://tv:pw@tvh.dkr:9981/stream/channel/065fa7ac302ba277a71b5f9a0168a30
```

so is this some new changes or features in the master version of tvheadend that isn't of the long term stable version? for example i noticed in the configuration on the server side some kind of passcode feature here:

config --> users --> passwords tab
select user --> click 'edit'

"Persistent authentication:"
Tooltip hint: "Manage Persistent authentication for HTTP streaming"
[ ] Enable
[ ] Reset

and it generates a long passcode, for example `Pj7oHMPu55rUQOxhG8u4KalNw6SM`
however i do not wish to use that feature, because it generates too long passwords for entering into a mobile device. and i definately do not understand how that feature is supposed to work. it was not enabled at the beginning of this troubles. so i switched on. but then switched it back off again. after seeing it did not change the 403 error msg

what else? well i noticed there is a 'web root' option in the server settings page in your app. and it is set to `/` which seems to be correct for my server.

what else?

well i noticed that my user is regular user (not admin). so under the webroot setting i turned 'off' the default admin on switch in the config page for the server settings. then next time i try to open a channel with internal vlc for playback... it now keeps asking me to login again over HTTP auth. And i must type the username + password again.

this seems unnecessary, and like some bug to me. because the app already has stored my password. its been configured

what else?

well on server side general settings. I noticed that on 'expert' level the password auth is set to 'md5sum' only. so the other option is 'plaintext'. or 'both. So i switched it to 'both'. to try to get rid of the new password nag screen. however it didnt/ and playback still getting the same 403 error

but regardless of those other bug(s) it will still after authentication return me a `403` missing page error. When trying to playback the channel stream

And its same for internal or external vlc. But the ios player it's just using maybe different type of API for that. because the server log was different msg

Out of ideas now. But happy to come back later if you think there is something else concrete i should be trying out here. Many thanks for this app BTW. Although I do not use it personally myself (it is for a family member)

**Comment by zipleen** [comment]
client="AppleCoreMedia/1.0.0.19C63 (iPad; U; CPU OS 15_2_1 like Mac OS X; en_gb)"

this means you are using "iOS player" option in settings, and that only works with transcoding to a specific stream supported by iOS, nothing to do with tvhclient. Try disabling this option, so VLC is used, which is what it should be.

also, if you want, get the latest bela which includes newer vlckit: https://testflight.apple.com/join/kNRWzusF

no code changes for tvheadend though! But you do need the setting to be in "both" if you ever want to have VLC playback. But you can try getting around that by disabling "admin" in the server config - this will fallback to token mode, but this only works automatically if the user has options to do it. If you remove the "it is admin", then this option is always used for playback, which "bypasses" the authentication method

**Comment by zipleen** [comment]
I guess maybe tvheadend latest git broke stuff? I'm using the latest stable version and having been on the latest bleeding edge git versions though. 
there's a discord with people active that talk about issues more in real time

**Comment by dreamcat4** [comment]
yeah i myself was also on the latest stable version for a very long time.

however last week i found i could compile the master branch, and since that is now the only ongoing active development (because nobody is tagging releases anymore over there...)

so now on master because other bug fixes or features / improvements. For example a feature was added to stop channels 'dissapearing by themselves'. that can be disabled now. Was my main reason for having to move over to the new master branch

thing is clearly (as you can guess), quite likely other things have gotten broken too. which might include specific API related to tvhclient app. sorry IDK the specifics to say what exactly it might be related to

however thanks for mentioning discord server. it might be a good place to discuss recent development. will try to join over there

**Comment by apfelsprudel** [comment]
Have the same here.
Try this: Hit settings / edit / Server settings / advanced settings / web root
Set something here, like a slash '/' and save.
Afterwards edit the server settings again and delete previous setting for web root and save.
This has resolved the issue for me and as I realized that on server side log tvhclient tries to access url /(null)/ which says that there is some value mismatch for "web root" setting.
